internal/runner/operation_task.go

```diff
@@ -180,3 +180,113 @@ func (r *Runner) executeStopTaskOp(
 
 	return &pb.Job_Result{}, nil
 }
+
+func (r *Runner) executeWatchTaskOp(
+	ctx context.Context,
+	log hclog.Logger,
+	ui terminal.UI,
+	job *pb.Job,
+) (*pb.Job_Result, error) {
+	op, ok := job.Operation.(*pb.Job_WatchTask)
+	if !ok {
+		// this shouldn't happen since the call to this function is gated
+		// on the above type match.
+		panic("operation not expected type")
+	}
+
+	// Look up the state from a start job.
+	startId := op.WatchTask.StartJob.Id
+	log = log.With("start-job-id", startId)
+	log.Debug("looking up start job to get state")
+	job, err := r.client.GetJob(ctx, &pb.GetJobRequest{
+		JobId: startId,
+	})
+
+	// If the job is not found, this is not an error. This means the
+	// start job never ran for whatever reason and we should not watch
+	// anything.
+	if status.Code(err) == codes.NotFound {
+		log.Warn("start job not found, skipping watch")
+		return nil, nil
+	} else if err != nil {
+		return nil, errors.Wrapf(err, "failed to look up job with id %s", startId)
+	}
+
+	// If the job is not in a terminal state, then its an error.
+	if job.State != pb.Job_SUCCESS && job.State != pb.Job_ERROR {
+		return nil, status.Errorf(codes.FailedPrecondition,
+			"cannot stop task when the start job is not terminal: %q",
```

Comment:
```suggestion
			"cannot watch task when the start job is not terminal: %q",
```

I think this is a copy/paste typo?